odoo/addons/base/wizard/base_partner_merge.py

```diff
@@ -193,10 +193,23 @@ def update_records(model, src, field_model='model', field_id='res_id'):
                 with mute_logger('odoo.sql_db'), self._cr.savepoint():
                     records.sudo().write({field_id: dst_partner.id})
                     records.env.flush_all()
-            except psycopg2.Error:
-                # updating fails, most likely due to a violated unique constraint
-                # keeping record with nonexistent partner_id is useless, better delete it
-                records.sudo().unlink()
+            except psycopg2.Error as e:
+                # updating fails, if the model of records is in allowed ones to delete in case
```

Comment:
This comment is hard to read IMHO.

I'd say something like:

```
If updating fails and record's model is in the ones allowed to...
```

Reply:
Done